boa/src/builtins/string/mod.rs

```diff
@@ -918,13 +918,17 @@ impl String {
             return Value::from(primitive);
         }
 
-        let filter = fill_string.as_deref().unwrap_or(" ");
+        let filler = fill_string.as_deref().unwrap_or(" ");
+
+        if filler == "" {
```

Comment:
I would write `filler.is_empty()`, it will be a bit more performant, and I think it's more clear when reading through the code.